crates/rust-analyzer/build.rs

```diff
@@ -18,29 +15,52 @@ fn set_rerun() {
     );
 
     while manifest_dir.parent().is_some() {
-        if manifest_dir.join(".git/HEAD").exists() {
-            let git_dir = manifest_dir.join(".git");
-
-            println!("cargo:rerun-if-changed={}", git_dir.join("HEAD").display());
-            // current branch ref
-            if let Ok(output) =
-                Command::new("git").args(&["rev-parse", "--symbolic-full-name", "HEAD"]).output()
-            {
-                if let Ok(ref_link) = String::from_utf8(output.stdout) {
-                    println!("cargo:rerun-if-changed={}", git_dir.join(ref_link).display());
-                }
-            }
+        let head_ref = manifest_dir.join(".git/HEAD");
+        if head_ref.exists() {
+            println!("cargo:rerun-if-changed={}", head_ref.display());
             return;
         }
 
         manifest_dir.pop();
     }
+
     println!("cargo:warning=Could not find `.git/HEAD` from manifest dir!");
 }
 
-fn rev() -> Option<String> {
-    let output =
-        Command::new("git").args(&["describe", "--tags", "--exclude", "nightly"]).output().ok()?;
+fn rev() -> String {
+    if let Ok(rev) = env::var("RUST_ANALYZER_REV") {
+        return rev;
+    }
+
+    if let Some(commit_hash) = commit_hash() {
+        let mut buf = commit_hash;
+
+        if let Some(date) = build_date() {
+            buf.push(' ');
+            buf.push_str(&date);
+        }
+
+        let channel = env::var("RUST_ANALYZER_CHANNEL").unwrap_or_else(|_| "dev".to_string());
+        buf.push(' ');
+        buf.push_str(&channel);
+
+        return buf;
+    }
+
+    "???????".to_string()
+}
+
+fn commit_hash() -> Option<String> {
+    output_to_string("git rev-parse --short HEAD")
+}
+
+fn build_date() -> Option<String> {
+    output_to_string("date --iso --utc")
```

Comment:
Not sure if this'll work for all hosts, but we are doing a similar thing in dist anyways, so, somehow, it seems to work?